sky/provision/kubernetes/network.py

```diff
@@ -0,0 +1,221 @@
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+from sky.provision import common
+from sky.provision.kubernetes import network_utils
+from sky.utils import kubernetes_enums
+from sky.utils import ux_utils
+from sky.utils.resources_utils import port_ranges_to_set
+
+_PATH_PREFIX = "/skypilot/{cluster_name_on_cloud}/{port}"
+_LOADBALANCER_SERVICE_NAME = "{cluster_name_on_cloud}-skypilot-loadbalancer"
+
+
+def _get_port_mode(
+        provider_config: Dict[str, Any]) -> kubernetes_enums.KubernetesPortMode:
+    mode_str = provider_config.get(
+        "port_mode", kubernetes_enums.KubernetesPortMode.LOADBALANCER.value)
+    try:
+        port_mode = kubernetes_enums.KubernetesPortMode.from_str(mode_str)
+    except ValueError as e:
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(str(e) + ' Cluster was setup with invalid port mode.'
+                              + 'Please check the port_mode in provider config.') \
+                from None
+
+    return port_mode
+
+
+def open_ports(
+    cluster_name_on_cloud: str,
+    ports: List[str],
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> None:
+    """See sky/provision/__init__.py"""
+    assert provider_config is not None, 'provider_config is required'
+    port_mode = _get_port_mode(provider_config=provider_config)
+    ports = list(port_ranges_to_set(ports))
+    if port_mode == kubernetes_enums.KubernetesPortMode.LOADBALANCER:
+        _open_ports_using_loadbalancer(
+            cluster_name_on_cloud=cluster_name_on_cloud,
+            ports=ports,
+            provider_config=provider_config)
+    elif port_mode == kubernetes_enums.KubernetesPortMode.INGRESS:
+        _open_ports_using_ingress(cluster_name_on_cloud=cluster_name_on_cloud,
+                                  ports=ports,
+                                  provider_config=provider_config)
+
+
+def _open_ports_using_loadbalancer(
```

Comment:
We should probably document the funny behavior of ELB health checks on EKS, where if a range of ports is opened, then the service won't work unless _all_ ports are backed by a service. We can note it as a TODO and in the docs.